### PR TITLE
New style/bessel

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/bessel.hpp [hunk 1]
```diff
@@ -20,4 +20,25 @@
...
+#include <eve/module/bessel/regular/cyl_bessel_y0.hpp>
+#include <eve/module/bessel/regular/cyl_bessel_y1.hpp>
+#include <eve/module/bessel/regular/cyl_bessel_yn.hpp>
+#include <eve/module/bessel/regular/sph_bessel_j0.hpp>
+#include <eve/module/bessel/regular/sph_bessel_j1.hpp>
+#include <eve/module/bessel/regular/sph_bessel_jn.hpp>
+#include <eve/module/bessel/regular/sph_bessel_y0.hpp>
+#include <eve/module/bessel/regular/sph_bessel_y1.hpp>
+#include <eve/module/bessel/regular/sph_bessel_yn.hpp>
+//#include <eve/module/bessel/regular/bessel.hpp>
```
Comment (jfalcou) [inline include/eve/module/bessel.hpp:44]: remove don't comment

File: include/eve/module/bessel/regular/impl/airy.hpp [hunk 2]
```diff
@@ -16,61 +16,60 @@
...
-  {
-    if( x == minf(as(x)) ) return kumi::tuple<T, T> {zero(as(x)), zero(as(x))};
-    if( is_nan(x) ) return kumi::tuple<T, T> {nan(as(x)), nan(as(x))};
-    if( x == inf(as(x)) ) return kumi::tuple<T, T> {zero(as(x)), inf(as(x))};
-    if( x < 0 ) return br_0();
-    if( x < thresh ) return br_small();
-    return br_last();
-  }
-  else
+  template<floating_ordered_value T, callable_options O>
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/impl/airy.hpp:19]: should be typename ?

File: include/eve/module/bessel/regular/sph_bessel_j0.hpp [hunk 3]
```diff
@@ -49,7 +62,14 @@ namespace eve
...
-}
+  inline constexpr auto sph_bessel_j0 = functor<sph_bessel_j0_t>;
 
-#include <eve/module/bessel/regular/impl/sph_bessel_j0.hpp>
+  namespace detail
+  {
+    template<typename T, callable_options O>
+    EVE_FORCEINLINE T sph_bessel_j0_(EVE_REQUIRES(cpu_), O const&, T x)
+    {
+       return sinc(x);
```
Comment (jfalcou) [inline include/eve/module/bessel/regular/sph_bessel_j0.hpp:72]: uh ? that's all ?

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/bessel/regular/sph_bessel_j0.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/bessel/regular/sph_bessel_j0.hpp:72]: yes

File: include/eve/module/bessel/regular/impl/airy.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/bessel/regular/impl/airy.hpp:19]: yes

File: include/eve/module/bessel.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/bessel.hpp:44]: it seems to me that this file has already been updated

**Review by jfalcou — APPROVED** [review]

**Pull request merged**